src/main/java/aeneas/models/Level.java

```diff
@@ -30,6 +31,9 @@ public void setRefreshListener(RefreshListener listener) {
   transient int levelNumber;
   transient boolean active = false;
   boolean prebuilt;
+  
+  transient Stack<IMove> undoStack;
```

Comment:
Also, what is the justification for having the undo/redo information in the levels, rather than the Model? I am willing to go along with it, but it is exceedingly awkward due to the fact that undo/redo moves could swap out the Level object itself.